pkg/util/pool/fast_releasing_pool_test.go

```diff
@@ -0,0 +1,205 @@
+package pool
+
+import (
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFastReleasingSlabPool(t *testing.T) {
+	t.Run("byte slices do not overlap when fit on the same slab", func(t *testing.T) {
+		delegatePool := &TrackedPool{Parent: &sync.Pool{}}
+		slabPool := NewFastReleasingSlabPool[byte](delegatePool, 10)
+
+		sliceA, slabIDA := slabPool.Get(5)
+		require.Len(t, sliceA, 5)
+		require.Equal(t, 5, cap(sliceA))
+		copy(sliceA, "12345")
+
+		sliceB, slabIDB := slabPool.Get(5)
+		require.Len(t, sliceB, 5)
+		require.Equal(t, 5, cap(sliceB))
+		copy(sliceB, "67890")
+
+		require.Equal(t, "12345", string(sliceA))
+		require.Equal(t, "67890", string(sliceB))
+
+		require.Equal(t, 2, len(slabPool.slabs)) // first slab (id=0) is always nil
+		require.Nil(t, slabPool.slabs[0])
+		require.Equal(t, 2, slabPool.slabs[1].references)
+		require.Equal(t, 10, slabPool.slabs[1].nextFreeIndex)
+
+		slabPool.Release(slabIDA)
+
+		require.Equal(t, 2, len(slabPool.slabs))
+		require.Nil(t, slabPool.slabs[0])
+		require.Equal(t, 1, slabPool.slabs[1].references)
+		require.Equal(t, 10, slabPool.slabs[1].nextFreeIndex)
+
+		slabPool.Release(slabIDB)
+
+		require.Equal(t, 2, len(slabPool.slabs))
+		require.Nil(t, slabPool.slabs[0])
+		require.Nil(t, slabPool.slabs[1])
+
+		// Allocating another slice needs a new slab.
+		sliceC, slabIDC := slabPool.Get(5)
+		require.Len(t, sliceC, 5)
+		require.Equal(t, 5, cap(sliceB))
```

Comment:
Did you mean sliceC here?

```suggestion
		require.Equal(t, 5, cap(sliceC))
```

Reply:
Thanks, fixed.